### PR TITLE
Save content-asset association during the index process

### LINKED ISSUE #1011: Save content-asset association during the index process
### Story

https://github.com/magento/adobe-stock-integration/issues/717 [Story 43] User sees pages the image is used in Magento

### Preconditions (*)
Install magneto and adobe stock integration

### Steps to reproduce (*)
Run `bin/magento media-gallery:sync`

### Expected result (*)
The content-asset association should be saved to `media_content_asset` table

### Has dependencies:
https://github.com/magento/adobe-stock-integration/issues/1147

### Related Issues:
https://github.com/magento/adobe-stock-integration/issues/964
https://github.com/magento/adobe-stock-integration/issues/1004

### PR DESCRIPTION
**Pull request by VladimirZaets**
### Description (*)
Synchronize content - asset links

### Related Pull Requests
https://github.com/magento/magento2/pull/27948
https://github.com/magento/partners-magento2ee/pull/236
https://github.com/magento/partners-magento2-infrastructure/pull/1

### Fixed Issues (if relevant)
https://github.com/magento/adobe-stock-integration/issues/1011

### CODE REVIEW

File: MediaContentSynchronizationApi/Model/SynchronizerPool.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    /**
+     * @param SynchronizeInterface[] $synchronizers
+     */
+    public function __construct(
+        array $synchronizers = []
+    ) {
+        foreach ($synchronizers as $synchronizer) {
+            if (!$synchronizer instanceof SynchronizeInterface) {
+                throw new \InvalidArgumentException(__('Synchronizer doesn\'t implement SynchronizeInterface'));
```
Comment (sivaschenko) [inline MediaContentSynchronizationApi/Model/SynchronizerPool.php]: Can you please also add this check to MediaGallery synchronizer pool for consistency

File: MediaContentSynchronization/Model/Synchronize.php [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+                $synchronizer->execute();
+            } catch (\Exception $exception) {
+                $this->log->critical($exception);
+                $failedItems[] = $exception;
+            }
+        }
+
+        if (!empty($failedItems)) {
+            throw new LocalizedException(
+                __('Could not synchronize content with asset')
```
Comment (sivaschenko) [inline MediaContentSynchronization/Model/Synchronize.php]: I would be good to provide more information on failed operation, at least failed synchronizer name

**Comment by sivaschenko** [comment]
@magento run all tests

**Comment by sivaschenko** [comment]
@magento run all tests

**Comment by sivaschenko** [comment]
https://m2build-comm-2.devops.magento.com/job/MTS-API-Tests/10129/

**Comment by Nazar65** [comment]
:heavy_check_mark: QA Passed

**Review by lenaorobei — DISMISSED** [review]

**Pull request merged**